acceptable/__main__.py

```diff
@@ -195,24 +211,57 @@ def parse_args(raw_args=None, parser_cls=None, stdin=None):
 
 
 def metadata_cmd(cli_args):
-    import_metadata(cli_args.modules)
+    import_metadata(cli_args.modules, cli_args.dummy_dependencies)
     current, _ = get_metadata().serialize()
-    print(json.dumps(current, indent=2))
+    cli_args.output.write(json.dumps(current, indent=2))
 
 
-def import_metadata(module_paths):
-    """Import all the given modules"""
+def add_working_dir_to_python_path():
     cwd = os.getcwd()
     if cwd not in sys.path:
         sys.path.insert(0, cwd)
-    modules = []
-    try:
-        for path in module_paths:
-            modules.append(import_module(path))
-    except ImportError as e:
-        err = RuntimeError('Could not import {}: {}'.format(path, str(e)))
-        raise_from(err, e)
-    return modules
+
+
+def import_or_exec(path):
+    """If path exists and ends in .py it is exec'd otherwise
+    import is attempted"""
+    if os.path.exists(path) and path.endswith('.py'):
+        try:
+            exec(open(path).read(), {}, {})
```

Comment:
Acceptable reads API metadata at import time into a global variable, acceptable._service._metadata. If we exec these python files with no global dict, does that still work? I guess it still uses the global sys.modules?

Reply:
It does work but its a good point - why does it work - I will take a look.

Reply:
Sorry I misunderstood you're question.

When using the dummy importer we do `import acceptable._service` @ __main__.py:244 to ensure it is in the interpreters module dict (exposed as sys.modules) before we enter the dummy context. The dummy context keeps modules that have already been imported in sys.modules but it makes a copy to so it can restore it afterwards. If we didn't do this real code could import the cached mock modules. 

We could add more control around this but we don't need it and it would add complexity.  

Running with a fresh globals doesn't effect the behaviour of importing or the interpreters module dict, its just like executing the code in a new module context.